app.rb

```diff
@@ -0,0 +1,188 @@
+require './book'
+require './person'
+require './rental'
+require './student'
+require './teacher'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def check_length(list, list_name)
+    return unless list.empty?
+
+    puts "There's no #{list_name} stored yet"
+    display_menu
+  end
+
+  def display_menu
+    menu = {
+      1 => 'List all books',
+      2 => 'List all people',
+      3 => 'Create a person',
+      4 => 'Create a book',
+      5 => 'Create a rental',
+      6 => 'List all rentals for a given person id',
+      7 => 'Exit'
+    }
+
+    puts "\nSelect an option by entering a number:"
+
+    menu.each do |key, value|
+      puts "#{key} - #{value}"
+    end
+
+    validate_option
+  end
+
+  def validate_option
+    option = gets.chomp.to_i
+
+    menu_actions = {
+      1 => :list_all_books,
+      2 => :list_all_people,
+      3 => :create_person,
+      4 => :create_book,
+      5 => :create_rental,
+      6 => :list_rentals_by_person_id,
+      7 => :exit
+    }
+
+    action = menu_actions[option]
+
+    if action
+      send(action)
+    else
+      puts 'Select a valid option'
+      display_menu
+    end
+  end
+
+  def list_all_people
+    check_length(@people, 'people')
+
+    @people.each do |person|
+      puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+
+    display_menu
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    @people << Student.new(age, nil, name)
+    puts 'New person (Student) created successfully'
+
+    display_menu
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    @people << Teacher.new(age, specialization, name)
```

Comment:
- [ ] when I try to add teacher I have an error about number of arguments, kindly fix this 
![image](https://github.com/ghreza-crypto/school-library/assets/66097371/1a574c70-abea-4d88-9cb9-740582c03d26)